subsys/settings/src/settings_shell.c

```diff
@@ -114,11 +114,46 @@ static int cmd_read(const struct shell *shell_ptr, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_write(const struct shell *shell_ptr, size_t argc, char *argv[])
+{
+	int err;
+	uint8_t buffer[CONFIG_SHELL_CMD_BUFF_SIZE / 2];
+	size_t buffer_len;
+
+	buffer_len = hex2bin(argv[2], strlen(argv[2]), buffer, sizeof(buffer));
+
+	if (buffer_len == 0) {
+		err = -ENOMEM;
+	} else {
+		err = settings_save_one(argv[1], buffer, buffer_len);
+	}
+
+	if (err) {
+		shell_error(shell_ptr, "Failed to write value: %d", err);
+	}
+
+	return 0;
+}
+
+static int cmd_delete(const struct shell *shell_ptr, size_t argc, char *argv[])
+{
+	int err;
+
+	err = settings_delete(argv[1]);
+
+	if (err) {
+		shell_error(shell_ptr, "Failed to delete value: %d", err);
```

Comment:
"Failed to delete key:" ?

Reply:
We delete both key and value, so it's probably more precise to say "delete setting". I'll change that.

Reply:
Done.